### DISCUSSION TITLE
How to add my own changes?

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [prasanthrangan/hyprdots#589] [Q&A]
The wiki says to add your own custom changes in `userprefs.conf`,but there are some keybindings that i wanted to change and if i just add them in `userprefs.conf`, both keybing gets run - from `keybindings.conf` and `userprefs.conf`. So i have to first comment the keybinding in `keybindings.conf`.

Now if i update the default gets overwritten with my settings.

So how to add my own custom changes?

What i think i would have to do is, either fork/or maitain a local branch and to --rebase while using --git pull.

is this the way? Any other way?

**Comment by kRHYME7** [comment]
Hello I Made a script to unbind conflicting keybindings so if you put your  custom keybindings inside userprefs.conf then run the script it will unbind the default keybindings. If that sounds alright to you I will try to issue a PR and I hope you'll try it.


You could do this manually using unbind = mod , key so at the top of your userprefs unbind all keybinding that might conflict


![code](https://github.com/prasanthrangan/hyprdots/assets/53417443/29c0855a-67df-4b50-8ffe-9cdd4a27405c)

**Reply by kRHYME7** [reply]
#600

**Reply by KAGEYAM4** [reply]
nice work.

**Comment by kRHYME7** [comment]
also forking is the way to go for now . It's a good way to know how the Dots work

**Reply by akashgagda** [reply]
Bro you use nix?

**Reply by kRHYME7** [reply]
Just nix package +  home manager but I only use that for testing stuff.
